ambari-server/src/main/java/org/apache/ambari/server/controller/metrics/MetricsPaddingMethod.java

```diff
@@ -65,6 +65,11 @@ public void applyPaddingStrategy(TimelineMetric metric, TemporalInfo temporalInf
     long dataStartTime = longToMillis(values.firstKey());
     long dataEndTime = longToMillis(values.lastKey());
 
+    // in the case we have invalid end time, replace it with last available timestamp
+    if (intervalEndTime < 0) {
```

Comment:
start < end is verified pre-condition already?

Reply:
Looks like they are pre-defined values:
https://github.com/apache/ambari/blob/219f22345e6f7b7ad38cdaac02e9bbe8c8ad7254/ambari-server/src/main/java/org/apache/ambari/server/api/services/BaseRequest.java#L209
And I didn't find that they were validated any-where before.

Reply:
Yes actually AMS also does validation, @avijayanhwx can point out where we do that but it's ok to verify here as well that since (start < end) should be a precondition check.

Reply:
So should I move the check to another place or it is ok?

Reply:
@dvitiiuk   nah, please add "start < end" check here as well

Reply:
AMS validation does happen, but this issue is due to infinite loop in the padding done in Ambari server after the metrics are got from AMS. I am +1 to add start < end check. Other than that, I am OK with the changes.